OSSM-8701: List recommendations for migration (#211)
openshift-service-mesh/sail-operator: docs/ossm/ossm2-migration/possible-problems-list/README.md

# OpenShift Service Mesh 2 --> 3 Migration risks and recommendations
Given the nature of the migration (not upgrading the operator but migrating to brand new operator), there are a few problems which can't be handled by any of the OpenShift Service Mesh 2 or OpenShift Service Mesh 3 operators. We are listing those problems and recommendations here so users can prepare in advance.

## Recommendations
Following are recommendations to keep the risk of misconfigurations or possible conflicts to minimum.

1. Do not upgrade OpenShift Service Mesh 2 operator or control planes in the middle of the migration

    After the OpenShift Service Mesh 3 operator is installed and migration of data plane is in progress, it's not recommended to upgrade OpenShift Service Mesh 2 operator or control planes. This can be achieved by switching from `Automatic` Operator Update approval to `Manual`.
1. Keep the amount of the service mesh configuration changes to minimum during the migration

>     It's not recommended to change Istio resources for traffic management or security or add new workloads to the service mesh in the middle of the migration.

Reviewer: ```suggestion
    It's not recommended to change Istio resources for traffic management or security or add new workloads or gateways to the service mesh in the middle of the migration.
```

Reviewer: Should we recommend that if they have to add new workloads and/or gateways that they add them to the OSSM 3 control plane and not the OSSM 2 control plane?

Author: They would have to still add the `maistra.io/ignore-namespace: "true"` label to that namespace which could be easy to miss. I will add a note.

Author: @nrfox PTAL https://github.com/openshift-service-mesh/sail-operator/pull/211/commits/2e77b3bb731d9ae07388fff3f302d3601cd8e511

## Recommendations
Following are recommendations to keep the risk of misconfigurations or possible conflicts to minimum.

1. Do not upgrade OpenShift Service Mesh 2 operator or control planes in the middle of the migration

    After the OpenShift Service Mesh 3 operator is installed and migration of data plane is in progress, it's not recommended to upgrade OpenShift Service Mesh 2 operator or control planes. This can be achieved by switching from `Automatic` Operator Update approval to `Manual`.
1. Keep the amount of the service mesh configuration changes to minimum during the migration

    It's not recommended to change Istio resources for traffic management or security or add new workloads or gateways to the service mesh in the middle of the migration.
    > **_NOTE:_** If it's necessary to add a new workload namespace during the migration, it should be managed by 3.0 control plane and it MUST be labeled with `maistra.io/ignore-namespace: "true"` to avoid conflicts between 3.0 and 2.6 control planes.
1. Finish the migration without unnecessary delays

>     To easier follow the recommendations above, when started the migration should be finished as soon as possible without unnecessary delays.

Reviewer 2: nit (if you happen to respin):

```suggestion
Once the migration is started, it should be completed as quickly as possible without unnecessary delays.
```

Author: Thanks
